drivers/unix/os_unix.cpp

```diff
@@ -166,7 +166,9 @@ void OS_Unix::initialize_core() {
 	DirAccess::make_default<DirAccessUnix>(DirAccess::ACCESS_USERDATA);
 	DirAccess::make_default<DirAccessUnix>(DirAccess::ACCESS_FILESYSTEM);
 
+#ifndef UNIX_SOCKET_UNAVAILABLE
 	NetSocketPosix::make_default();
+#endif
```

Comment:
Maybe `POSIX_SOCKET_UNAVAILABLE` would indeed be a better name.

I'm just reusing `UNIX_SOCKET_UNAVAILABLE` which was added in #32454 to allow building on some console platform.

I've actually been thinking about splitting Windows sockets since they've diverged quite a bit from the posix one (and their "compatibility" layer is [not compatible](https://daniel.haxx.se/blog/2012/10/10/wsapoll-is-broken/))

Reply:
Yeah splitting Windows sockets might make sense, I've thought about this too on and off.

Reply:
I've opened https://github.com/godotengine/godot/pull/98969 which does that.